tanjun/commands.py

```diff
@@ -1276,8 +1276,8 @@ def add_str_option(
             actual_choices = {}
             warned = False
             for choice in choices:
-                if isinstance(choice, tuple):
-                    if not warned:
+                if isinstance(choice, tuple):  # type: ignore[unreachable]  # the point of this is for deprecation
```

Comment:
`choice` is not possibly a `tuple`, which is why mypy (correctly) complains here: that's alright, because this code will be removed soon; it's deprecated.